xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/servletEngine/ServletContainerExecutor.java

```diff
@@ -170,6 +173,77 @@ private void startContainer() throws Exception
         start(this.servletContainer, this.testConfiguration);
     }
 
+    private void throwExceptionForServletEngineNotYetSupported(TestConfiguration testConfiguration)
+    {
+        throw new RuntimeException(String.format("Servlet engine [%s] is not yet supported!",
+            testConfiguration.getServletEngine()));
+    }
+
+    private GenericContainer getServletContainer(TestConfiguration testConfiguration)
+    {
```

Comment:
I don't like this method. We already have a switch above and I don't think it's good to repeat it. Makes it harder to read (since the logic is split into 2 places). Better to have methods or classes based on servlet engines IMO.